fish 3.1.0 - build - fatal error: curses.h: No such file or directory

fish: 3.1.0
cmake: 3.14.1
ncurses: 5.9

I have no local /usr installation of ncurses with development files.  It is in a non-standard location.

If I set PKG_CONFIG_PATH to have the ncurses pkgconfig dir, ncurses is not found.
I also tried setting PATH so ncurses5-config could be found.

So I set
```
  -DCMAKE_SYSTEM_PREFIX_PATH="${UTILS_NCURSES};${UTILS_GETTEXT};${UTILS_READLINE}"
```
And then ncurses is found in the path that I expect.

However, it does not seem that the include path gets added to the build, I encounter this error when building:
```
/build/tmp/fish-3.1.0/src/builtin_set_color.cpp:8:20: fatal error: curses.h: No such file or directory
```

So additionally I am setting CFLAGS/CXXFLAGS
```
 -DCMAKE_C_FLAGS="-O2 -I${UTILS_NCURSES}/include" \
 -DCMAKE_CXX_FLAGS="-O2 -I${UTILS_NCURSES}/include" \
```
(the files here in "include" are links to "include/ncursesw" files)

In the end the build looks something like:
```
export PATH=${UTILS_CMAKE}/bin:$PATH

export CC=${UTILS_GCC}/bin/gcc
export CXX=${UTILS_GCC}/bin/g++

export LDFLAGS+=" -static-libgcc -static-libstdc++"
export LDFLAGS+=" -Wl,-rpath,${UTILS_NCURSES}/lib"

mkdir build && cd build
cmake \
 -DCMAKE_BUILD_TYPE=Release \
 -DCMAKE_INSTALL_PREFIX=${PREFIX} \
 -DCMAKE_SYSTEM_PREFIX_PATH="${UTILS_NCURSES};${UTILS_GETTEXT};${UTILS_READLINE}" \
 -DCMAKE_C_FLAGS="-O2 -I${UTILS_NCURSES}/include" \
 -DCMAKE_CXX_FLAGS="-O2 -I${UTILS_NCURSES}/include" \
..

make -j${nproc}
make install
```

But ideally I would not have to set the CFLAGS/CXXFLAGS.

I am running this in a CentOS 6 docker container.

This is the output when I run with pkg-config
```
export PKG_CONFIG_PATH=${UTILS_NCURSES}/lib/pkgconfig

mkdir build && cd build
cmake \
 -DCMAKE_BUILD_TYPE=Release \
 -DCMAKE_INSTALL_PREFIX=${PREFIX} \
..
```

```
-- The C compiler identification is GNU 5.5.0
-- The CXX compiler identification is GNU 5.5.0
-- Check for working C compiler: /home/utils/gcc-5.5.0/bin/gcc
-- Check for working C compiler: /home/utils/gcc-5.5.0/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /home/utils/gcc-5.5.0/bin/g++
-- Check for working CXX compiler: /home/utils/gcc-5.5.0/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Could NOT find Curses (missing: CURSES_LIBRARY CURSES_INCLUDE_PATH)
Falling back to pkg-config for (n)curses detection
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.23")
-- Checking for one of the modules 'ncurses;curses'
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
```
...

```
-- The following OPTIONAL packages have not been found:

 * Curses
 * BZip2
 * Readline
 * Editline
```

I do see this in CMakeCache.txt, so it looks like pkgconfig finds ncurses.
```
pkgcfg_lib_CURSES_ncurses:FILEPATH=/home/utils/ncurses-5.9-2/lib/libncurses.so

//Path to a library.
pkgcfg_lib_CURSES_tinfo:FILEPATH=/home/utils/ncurses-5.9-2/lib/libtinfo.so
```

And in CMakeErrors.log I see:
```
Determining if the include file ncurses/term.h exists failed with the following output:
Change Dir: /build/tmp/fish-3.1.0/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/gmake cmTC_1b614/fast
/usr/bin/gmake -f CMakeFiles/cmTC_1b614.dir/build.make CMakeFiles/cmTC_1b614.dir/build
gmake[1]: Entering directory `/build/tmp/fish-3.1.0/build/CMakeFiles/CMakeTmp'
Building CXX object CMakeFiles/cmTC_1b614.dir/CheckIncludeFile.cxx.o
/home/utils/gcc-5.5.0/bin/g++  -D_GNU_SOURCE=1  -Wall -Wextra -Wno-implicit-fallthrough -Wno-comment -Wno-address -Wno-strict-aliasing -Wno-redundant-move     -std=gnu++11 -o CMakeFiles/cmTC_1b614.dir/CheckIncludeFile.cxx.o -c /build/tmp/fish-3.1.0/build/CMakeFiles/CMakeTmp/CheckIncludeFile.cxx
/build/tmp/fish-3.1.0/build/CMakeFiles/CMakeTmp/CheckIncludeFile.cxx:1:26: fatal error: ncurses/term.h: No such file or directory
compilation terminated.
gmake[1]: *** [CMakeFiles/cmTC_1b614.dir/CheckIncludeFile.cxx.o] Error 1
gmake[1]: Leaving directory `/build/tmp/fish-3.1.0/build/CMakeFiles/CMakeTmp'
gmake: *** [cmTC_1b614/fast] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -177,6 +177,8 @@ target_sources(fishlib PRIVATE ${FISH_HEADERS})
 target_link_libraries(fishlib
   ${CURSES_LIBRARY} ${CURSES_EXTRA_LIBRARY} Threads::Threads ${CMAKE_DL_LIBS}
   ${PCRE2_LIB} ${Intl_LIBRARIES} ${ATOMIC_LIBRARY})
+target_include_directories(fishlib PRIVATE
+  ${CURSES_INCLUDE_DIRS})
 
 # Define fish.
 add_executable(fish src/fish.cpp)

--- a/cmake/ConfigureChecks.cmake
+++ b/cmake/ConfigureChecks.cmake
@@ -32,6 +32,8 @@ if(NOT ${CURSES_FOUND})
     set(CURSES_CURSES_LIBRARY ${CURSES_LIBRARIES})
     set(CURSES_LIBRARY ${CURSES_LIBRARIES})
 endif()
+# Set up extra include directories for CheckIncludeFile
+list(APPEND CMAKE_REQUIRED_INCLUDES ${CURSES_INCLUDE_DIRS})
 
 # Fix undefined reference to tparm on RHEL 6 and potentially others
 # If curses is found via CMake, it also links against tinfo if it exists. But if we use our

